src/python/pants/backend/awslambda/python/target_types.py

```diff
@@ -106,13 +110,18 @@ class PythonAwsLambdaRuntime(PythonFaaSRuntimeField):
     # https://gallery.ecr.aws/lambda/python
     known_runtimes_docker_repo = "public.ecr.aws/lambda/python"
     known_runtimes = (
-        PythonFaaSKnownRuntime(3, 6, "3.6"),
-        PythonFaaSKnownRuntime(3, 7, "3.7"),
-        PythonFaaSKnownRuntime(3, 8, "3.8-x86_64"),
-        PythonFaaSKnownRuntime(3, 9, "3.9-x86_64"),
-        PythonFaaSKnownRuntime(3, 10, "3.10-x86_64"),
-        PythonFaaSKnownRuntime(3, 11, "3.11-x86_64"),
-        PythonFaaSKnownRuntime(3, 12, "3.12-x86_64"),
+        PythonFaaSKnownRuntime(3, 6, "3.6", AWSLambdaArchitecture.X86_64),
+        PythonFaaSKnownRuntime(3, 7, "3.7", AWSLambdaArchitecture.X86_64),
+        PythonFaaSKnownRuntime(3, 8, "3.8-x86_64", AWSLambdaArchitecture.X86_64),
+        PythonFaaSKnownRuntime(3, 8, "3.8-arm64", AWSLambdaArchitecture.ARM64),
```

Comment:
Just confirming, adding these new ones like this does the right thing automatically with `build-support/bin/generate_faas_complete_platforms.py` and that's how you generated the new `.json` files?

Reply:
Nope, I generated the new files using the Docker base image that Lambda provides, and then following your guide from https://github.com/pantsbuild/pants/discussions/18756. I'll try the script and make sure that it works with the new setup.

Reply:
I fixed the `generate_faas_complete_platforms.py` to work with both x86 and ARM images; however, this requires that the underlying Docker engine has `buildx` installed and that the system has `qemu` installed so that both types of images can be run.

I also ran into a bit of a catch-22 when using `pants run` to run this script as the files output by the script trigger the file-watcher to restart it; I've added comments that highlight both of these issues to the top of the script.